pkg/stanza/fileconsumer/internal/fingerprint/fingerprint.go

```diff
@@ -71,13 +86,29 @@ func (f Fingerprint) Equal(other *Fingerprint) bool {
 // a fingerprint. As the file grows, its fingerprint is updated
 // until it reaches a maximum size, as configured on the operator
 func (f Fingerprint) StartsWith(old *Fingerprint) bool {
-	l0 := len(old.FirstBytes)
-	if l0 == 0 {
+	lenOld := old.NBytesUsed
+	if lenOld == 0 {
 		return false
 	}
-	l1 := len(f.FirstBytes)
-	if l0 > l1 {
+	lenCurrent := len(f.firstBytes)
+	if lenOld > lenCurrent {
 		return false
 	}
-	return bytes.Equal(old.FirstBytes[:l0], f.FirstBytes[:l0])
+	if f.NBytesUsed == old.NBytesUsed {
+		return bytes.Equal(f.HashBytes, old.HashBytes)
+	}
+	h := fnv.New128a()
+	h.Write(f.firstBytes[:lenOld])
+	hash := h.Sum(nil)
+	return bytes.Equal(old.HashBytes, hash)
+}
+
+// IsMaxSize checks to see if fingerprint has reached its maxed size
+func (f Fingerprint) IsMaxSize(maxFingerprintSize int, offset int64) bool {
```

Comment:
I think the notion of an offset should not be a concern of the fingerprint itself. We can manage this externally to keep the api cleaner. Since we need to keep `FirstBytes` exported for backwards compatibility reasons, I think we can avoid adding this function for now, and should because it increases the surface area of a public package.